### PR TITLE
Fix flash attention

### PR DESCRIPTION
**Pull request by liamcli**
Thanks for maintaining this great open source library!  Quick fix for first contrib: fixing flash attention which is currently broken for micro batch size > 1 due to unstored reshape of `query_layer`

### CODE REVIEW

**Comment by liamcli** [comment]
Please let me know what I need to do to get this PR merged.

**Comment by StellaAthena** [comment]
> Please let me know what I need to do to get this PR merged.

Thank you for the PR submission. We are a bit behind but will review this when we can. Hopefully soon.

File: megatron/data/data_utils.py [hunk 1]
```diff
@@ -286,49 +286,72 @@ def weights_by_num_docs(l: list, alpha=0.3):
     return weights
 
 
-def build_train_valid_test_data_iterators(neox_args):
-    """XXX"""
+def build_datasets_from_neox_args(neox_args):
```
Comment (Quentin-Anthony) [inline megatron/data/data_utils.py:289]: Why are you modifying the dataloader?

File: megatron/neox_arguments/arguments.py [hunk 2]
```diff
@@ -359,16 +367,20 @@ def consume_deepy_args(cls):
...
-        for k, v in vars(args_parsed).items():
-            if k == "autotuning" and v is not None:
-                overwrite_values["autotuning_run"] = v
-            elif k not in ["conf_dir", "conf_file"] and v is not None:
-                overwrite_values[k] = v
+        possible_args = inspect.getfullargspec(cls)[0]
+        full_overwrite_values = dict()
+        for k, v in args_parsed.items():
+            if (
+                k not in ["conf_dir", "conf_file"]
```
Comment (Quentin-Anthony) [inline megatron/neox_arguments/arguments.py:374]: @dashstander -- Will removing the `overwrite_values["autotuning_run"]` break your autotuning code here?

Comment (dashstander) [reply megatron/neox_arguments/arguments.py:374]: Yes, that block is there for a reason. Did it absolutely need to go?

**Review by Quentin-Anthony — APPROVED** [review]
LGTM

**Pull request merged**